Source/Core/InputCommon/DynamicInputTextureConfiguration.cpp

```diff
@@ -0,0 +1,261 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include "InputCommon/DynamicInputTextureConfiguration.h"
+
+// Ignore opencv warnings
+#pragma warning(push, 0)
+#include <opencv2/core/core.hpp>
+#include <opencv2/features2d/features2d.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+#include <opencv2/imgproc/types_c.h>
+#include <opencv2/opencv.hpp>
+#pragma warning(pop)
+#include <picojson/picojson.h>
+
+#include <fstream>
+
+#include "Common/FileUtil.h"
+#include "Common/StringUtil.h"
+#include "Core/ConfigManager.h"
+#include "InputCommon/ControllerEmu/ControllerEmu.h"
+
+#include "VideoCommon/HiresTextures.h"
+#include "VideoCommon/TextureCacheBase.h"
+
+namespace
+{
+cv::Mat resizeKeepAspectRatio(const cv::Mat& input, const cv::Size& dstSize,
+                              const cv::Scalar& bgcolor)
+{
+  cv::Mat output;
+
+  double h1 = dstSize.width * (input.rows / (double)input.cols);
```

Comment:
`h1` and `w2` can be const.  There's a few other occurrences within the function. also, `static_cast`.